api/policies/v1/admissionpolicygroup_types.go

```diff
@@ -0,0 +1,200 @@
+	Status PolicyStatus             `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// AdmissionPolicyGroupList contains a list of AdmissionPolicyGroup.
+type AdmissionPolicyGroupList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []AdmissionPolicyGroup `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&AdmissionPolicyGroup{}, &AdmissionPolicyGroupList{})
+}
+
+func (r *AdmissionPolicyGroup) SetStatus(status PolicyStatusEnum) {
+	r.Status.PolicyStatus = status
+}
+
+func (r *AdmissionPolicyGroup) GetPolicyMode() PolicyMode {
+	return r.Spec.Mode
+}
+
+func (r *AdmissionPolicyGroup) SetPolicyModeStatus(policyMode PolicyModeStatus) {
+	r.Status.PolicyMode = policyMode
+}
+
+func (r *AdmissionPolicyGroup) IsContextAware() bool {
+	return false
+}
+
+func (r *AdmissionPolicyGroup) GetStatus() *PolicyStatus {
+	return &r.Status
+}
+
+func (r *AdmissionPolicyGroup) GetModule() string {
+	return ""
+}
+
+func (r *AdmissionPolicyGroup) GetPolicyMembers() []PolicyGroupMember {
+	return r.Spec.Policies
+}
+
+func (r *AdmissionPolicyGroup) IsPolicyGroup() bool {
+	return true
+}
+
+func (r *AdmissionPolicyGroup) GetSettings() runtime.RawExtension {
+	return runtime.RawExtension{}
+}
+
+func (r *AdmissionPolicyGroup) IsMutating() bool {
+	// By design, AdmissionPolicyGroup is always non-mutating.
+	// Policy groups can be used only for validating admission requests
+	return false
+}
+
+func (r *AdmissionPolicyGroup) GetExpression() string {
+	return r.Spec.Expression
+}
+
+func (r *AdmissionPolicyGroup) GetMessage() string {
+	return r.Spec.Message
+}
+
+func (r *AdmissionPolicyGroup) CopyInto(policy *Policy) {
+	*policy = r.DeepCopy()
+}
+
+func (r *AdmissionPolicyGroup) GetSideEffects() *admissionregistrationv1.SideEffectClass {
+	return r.Spec.SideEffects
+}
+
+// GetRules returns all rules. Scope is namespaced since AdmissionPolicyGroup just watches for namespace resources.
+func (r *AdmissionPolicyGroup) GetRules() []admissionregistrationv1.RuleWithOperations {
+	namespacedScopeV1 := admissionregistrationv1.NamespacedScope
+	rules := make([]admissionregistrationv1.RuleWithOperations, 0)
+	for _, rule := range r.Spec.Rules {
+		rule.Scope = &namespacedScopeV1
+		rules = append(rules, rule)
+	}
+
+	return rules
+}
+
+func (r *AdmissionPolicyGroup) GetFailurePolicy() *admissionregistrationv1.FailurePolicyType {
+	return r.Spec.FailurePolicy
+}
+
+func (r *AdmissionPolicyGroup) GetMatchPolicy() *admissionregistrationv1.MatchPolicyType {
+	return r.Spec.MatchPolicy
+}
+
+func (r *AdmissionPolicyGroup) GetMatchConditions() []admissionregistrationv1.MatchCondition {
+	return r.Spec.MatchConditions
+}
+
+// GetNamespaceSelector returns the namespace of the AdmissionPolicyGroup since it is the only namespace we want the policy to be applied to.
+func (r *AdmissionPolicyGroup) GetUpdatedNamespaceSelector(string) *metav1.LabelSelector {
+	return &metav1.LabelSelector{
```

Comment:
Shouldn't be exclude the `kubewarden` namespace as we do in ClusterAdmissionPolicyGroups?
I realize that here in AdmissionPolicyGroup we have the same behaviour as in AdmissionPolicies.

Still, I'm not sure this isn't a bug both in AdmissionPolicies and AdmissionPolicyGroup.

Reply:
Good question, I do not remember if this is done in purpose. But I would keep as it is now and maybe open an issue to fix it later.